readthedocs/projects/exceptions.py

```diff
@@ -1,15 +1,46 @@
 """Project exceptions"""
 
+from django.conf import settings
+from django.utils.translation import ugettext_noop as _
 
-class ProjectImportError (Exception):
+from readthedocs.doc_builder.exceptions import BuildEnvironmentError
 
-    """Failure to import a project from a repository."""
 
-    pass
+class ProjectConfigurationError(BuildEnvironmentError):
+
+    """Error raised trying to configure a project for build"""
+
+    NOT_FOUND = _(
+        'A configuration file was not found. '
+        'Make sure you have a conf.py file in your repository.'
+    )
+
+
+class RepositoryError(BuildEnvironmentError):
+
+    """Failure during repository operation."""
+
+    PRIVATE_REPO = _(
+        'There was a problem connecting to your repository, '
+        'ensure that your repository URL is correct.'
+    )
```

Comment:
If it's a private repo, shouldn't we warn them here? Why are these messages the same?

Reply:
We aren't thoroughly testing if this was a private repo or not, we'd have to grep the git output. @humitos raised a similar question and the best answer is to execute checkout in a build environment and report the commands, not parse them to determine what the error was.

Messages aren't the same, but usage could be clearer. I should probably rename these constants, as `PRIVATE_REPO` is raised if we support private repositories -- that is, we tell the user "check your url, because it wasn't the repository privacy that caused the problem. `PUBLIC_REPO` error is "check your url or your project privacy, because we don't support private repositories."

In fact, I'll update the copy here to be more explicit about repo privacy.